src/Adapter/Feature/CommandHandler/AddFeatureValueHandler.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Feature\CommandHandler;
+
+use FeatureValue;
+use PrestaShop\PrestaShop\Adapter\Feature\Repository\FeatureRepository;
+use PrestaShop\PrestaShop\Adapter\Feature\Repository\FeatureValueRepository;
+use PrestaShop\PrestaShop\Core\Domain\Feature\Command\AddFeatureValueCommand;
+use PrestaShop\PrestaShop\Core\Domain\Feature\CommandHandler\AddFeatureValueHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Feature\ValueObject\FeatureValueId;
+
+/**
+ * Handles adding of feature value using legacy model.
+ */
+class AddFeatureValueHandler implements AddFeatureValueHandlerInterface
```

Comment:
When we migrate Features BO page, we'll have issues 🤔  because from this BO page you can list, add, edit not only "features" entries but also "feature values" entries (the pre-defined values).

We should try to avoid confusion if possible.
Maybe we can rename (in the code) "feature values" into "features pre-defined values" for the other BO page and keep "feature values" here?

Reply:
Anyway, clearing the confusion around this data structure by using PHPDoc could be useful 👍 

```
/**
 * Handles adding of feature value using legacy model.
 *
 * Feature values are the merchant input configured for a given product for a given feature. Example: "Type of paper: Gloss"
 */
```

Reply:
I already added a comment in that sense in `AddFeatureValueCommand`

Reply:
So to clarify, by default a FeatureValue is a predefined value They are managed outside of the Product page in a list which is common to all products So a FeatureValue is predefined by default

The commands I added are meant for this purpose, the creation and edition of predefined FeatureValue it's the default behaviour so we shouldn't rename it (though I did add a comment in the `AddFeatureValueCommand` to clarify it a bit)

Reply:
By the way that's the reason why these commands are in their own `Feature` namespace and not in a sub namespace of `Product`

Reply:
Then in the product page you can dynamically create custom feature values, they will added on the fly on marked as `custom` This is the exception use case not the default one (hence the custom)

These custom feature values are handled by the `SetProductFeatureValuesCommand` and are in the scope of the product, they are attached to this product and won't be usable by any other product That's why when you go in the FeatureValue list page you don't see them, you only see pre-defined values